pkg/kubelet/status/status_manager.go

```diff
@@ -463,6 +465,25 @@ func (m *manager) TerminatePod(pod *v1.Pod) {
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.PodDisruptionConditions) {
+		// Make sure all deleted pods are transitioned to a terminal phase.
+		if pod.DeletionTimestamp != nil {
```

Comment:
If you remove this if, do the tests still pass?  I'm looking at the code now to make sure that when this is called the pod is definitely never going to get a new status update.  If so, then we don't need the if check.  If we *don't*, I may just fix that (because it would delay pod shutdown).

Reply:
If I deleted it there is one test that fails: https://github.com/kubernetes/kubernetes/blob/master/test/e2e_node/mirror_pod_test.go#L181. 

Here, the static pod is recreated, and the expectation is that the corresponding MirrorPod would continue to run, even though TerminatePod is called on the MirrorPod. 

Another solution would be to use `kubetypes.IsMirrorPod` to skip this, but I thought the check for deletion is more explicitly expressing the intention.

Reply:
Re: mirror pod - oh no, you found a bug in the kubelet!  This is a bad bug too for static pods, let me look at what kubelet is firing an error.  So this is great that the test exercises it.

I think #113145 when it merges should report better status here, let's wait until that lands and then rerun your tests on top of it.

Reply:
Ok, I just pushed the commit to do not use DeletionTimestamp, but needed to exclude mirror and static pods for the tests to pass.